From 39d0c8b866acba9be143c8014bff94e6c7d73131 Mon Sep 17 00:00:00 2001
From: Adarsh0127-Elite <adarshmoharana027@gmail.com>
Date: Wed, 30 Sep 2026 15:37:28 +0000
Subject: [PATCH 9/9] MiuiCamera: Unlock cloud watermarks and filters

- Bypass device model whitelist for cloud filter and watermark items
- Prevent deletion and filtering of downloaded cloud watermark resources

Signed-off-by: Adarsh0127-Elite <adarshmoharana027@gmail.com>
---
 smali_classes5/Rg/L.smali                     | 486 +-----------------
 .../cloudfilter/entity/CloudFilterItem.smali  |  59 +--
 2 files changed, 4 insertions(+), 541 deletions(-)

diff --git a/smali_classes5/Rg/L.smali b/smali_classes5/Rg/L.smali
index 9e839b8c..dc9319f6 100644
--- a/smali_classes5/Rg/L.smali
+++ b/smali_classes5/Rg/L.smali
@@ -234,491 +234,7 @@
 .end method
 
 .method public final d(Z)V
-    .locals 8
-
-    new-instance v0, Ljava/util/ArrayList;
-
-    invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
-
-    invoke-virtual {p0}, LRg/L;->f()Ljava/util/List;
-
-    move-result-object v1
-
-    new-instance v2, Ljava/lang/StringBuilder;
-
-    const-string v3, "filterData: E "
-
-    invoke-direct {v2, v3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {v2, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v1
-
-    iget-object v2, p0, LRg/L;->a:Ljava/lang/String;
-
-    invoke-static {v2, v1}, LWr/e;->m(Ljava/lang/String;Ljava/lang/String;)V
-
-    sget-object v1, LRg/s;->a:Lqv/n;
-
-    iget-object v1, p0, LRg/L;->h:LRg/s$a;
-
-    iget-object v3, p0, LRg/L;->c:Ljava/util/ArrayList;
-
-    invoke-virtual {p0}, LRg/L;->f()Ljava/util/List;
-
-    move-result-object p0
-
-    sget-object v4, LRg/s;->a:Lqv/n;
-
-    invoke-virtual {v4}, Lqv/n;->getValue()Ljava/lang/Object;
-
-    move-result-object v4
-
-    check-cast v4, Ljava/lang/Boolean;
-
-    invoke-virtual {v4}, Ljava/lang/Boolean;->booleanValue()Z
-
-    move-result v4
-
-    if-eqz v4, :cond_0
-
-    goto/16 :goto_f
-
-    :cond_0
-    invoke-interface {p0}, Ljava/util/List;->isEmpty()Z
-
-    move-result v4
-
-    if-eqz v4, :cond_1
-
-    goto :goto_1
-
-    :cond_1
-    if-eqz v3, :cond_2
-
-    invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object v4
-
-    :goto_0
-    invoke-interface {v4}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v5
-
-    if-eqz v5, :cond_2
-
-    invoke-interface {v4}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v5
-
-    check-cast v5, LRg/F;
-
-    iget-object v5, v5, LRg/F;->b:Ljava/util/ArrayList;
-
-    new-instance v6, LRg/u;
-
-    const/4 v7, 0x0
-
-    invoke-direct {v6, v7, p0, v0}, LRg/u;-><init>(ILjava/lang/Object;Ljava/lang/Object;)V
-
-    invoke-static {v5, v6}, Lrv/q;->A(Ljava/util/List;LFv/l;)V
-
-    goto :goto_0
-
-    :cond_2
-    :goto_1
-    invoke-static {}, Ljava/time/LocalDateTime;->now()Ljava/time/LocalDateTime;
-
-    move-result-object p0
-
-    if-eqz v3, :cond_3
-
-    invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object v4
-
-    :goto_2
-    invoke-interface {v4}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v5
-
-    if-eqz v5, :cond_3
-
-    invoke-interface {v4}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v5
-
-    check-cast v5, LRg/F;
-
-    iget-object v5, v5, LRg/F;->b:Ljava/util/ArrayList;
-
-    new-instance v6, LRg/B;
-
-    invoke-direct {v6, p0}, LRg/B;-><init>(Ljava/time/LocalDateTime;)V
-
-    invoke-static {v5, v6}, Lrv/q;->A(Ljava/util/List;LFv/l;)V
-
-    goto :goto_2
-
-    :cond_3
-    const/4 p0, 0x0
-
-    if-eqz v1, :cond_4
-
-    iget-object v4, v1, LRg/s$a;->a:Ljava/lang/String;
-
-    goto :goto_3
-
-    :cond_4
-    move-object v4, p0
-
-    :goto_3
-    if-nez v4, :cond_5
-
-    goto :goto_6
-
-    :cond_5
-    iget-object v4, v1, LRg/s$a;->a:Ljava/lang/String;
-
-    invoke-static {v4}, LGv/o;->e(Ljava/lang/Object;)V
-
-    sget-object v5, Ljava/util/Locale;->ROOT:Ljava/util/Locale;
-
-    invoke-virtual {v4, v5}, Ljava/lang/String;->toLowerCase(Ljava/util/Locale;)Ljava/lang/String;
-
-    move-result-object v4
-
-    const-string v5, "this as java.lang.String).toLowerCase(Locale.ROOT)"
-
-    invoke-static {v4, v5}, LGv/o;->g(Ljava/lang/Object;Ljava/lang/String;)V
-
-    const-string v5, "filterByDevice: currentDeviceName = "
-
-    invoke-virtual {v5, v4}, Ljava/lang/String;->concat(Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object v5
-
-    const-string v6, "LimitationUtil"
-
-    invoke-static {v6, v5}, LWr/e;->m(Ljava/lang/String;Ljava/lang/String;)V
-
-    if-eqz v3, :cond_6
-
-    invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object v5
-
-    :goto_4
-    invoke-interface {v5}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v6
-
-    if-eqz v6, :cond_6
-
-    invoke-interface {v5}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v6
-
-    check-cast v6, LRg/F;
-
-    iget-object v6, v6, LRg/F;->b:Ljava/util/ArrayList;
-
-    new-instance v7, LRg/v;
-
-    invoke-direct {v7, v4}, LRg/v;-><init>(Ljava/lang/String;)V
-
-    invoke-static {v6, v7}, Lrv/q;->A(Ljava/util/List;LFv/l;)V
-
-    goto :goto_4
-
-    :cond_6
-    if-eqz v3, :cond_7
-
-    invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object v5
-
-    :goto_5
-    invoke-interface {v5}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v6
-
-    if-eqz v6, :cond_7
-
-    invoke-interface {v5}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v6
-
-    check-cast v6, LRg/F;
-
-    iget-object v6, v6, LRg/F;->b:Ljava/util/ArrayList;
-
-    new-instance v7, LRg/w;
-
-    invoke-direct {v7, v4}, LRg/w;-><init>(Ljava/lang/String;)V
-
-    invoke-static {v6, v7}, Lrv/q;->A(Ljava/util/List;LFv/l;)V
-
-    goto :goto_5
-
-    :cond_7
-    :goto_6
-    if-eqz v1, :cond_8
-
-    iget-object v4, v1, LRg/s$a;->c:Ljava/lang/String;
-
-    goto :goto_7
-
-    :cond_8
-    move-object v4, p0
-
-    :goto_7
-    if-nez v4, :cond_9
-
-    goto :goto_9
-
-    :cond_9
-    if-eqz v3, :cond_a
-
-    invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object v4
-
-    :goto_8
-    invoke-interface {v4}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v5
-
-    if-eqz v5, :cond_a
-
-    invoke-interface {v4}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v5
-
-    check-cast v5, LRg/F;
-
-    iget-object v5, v5, LRg/F;->b:Ljava/util/ArrayList;
-
-    sget-object v6, LRg/t;->a:LRg/t;
-
-    invoke-static {v5, v6}, Lrv/q;->A(Ljava/util/List;LFv/l;)V
-
-    goto :goto_8
-
-    :cond_a
-    :goto_9
-    if-eqz v3, :cond_b
-
-    invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object v4
-
-    :goto_a
-    invoke-interface {v4}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v5
-
-    if-eqz v5, :cond_b
-
-    invoke-interface {v4}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v5
-
-    check-cast v5, LRg/F;
-
-    iget-object v5, v5, LRg/F;->b:Ljava/util/ArrayList;
-
-    new-instance v6, LRg/A;
-
-    invoke-direct {v6, v1}, LRg/A;-><init>(LRg/s$a;)V
-
-    invoke-static {v5, v6}, Lrv/q;->A(Ljava/util/List;LFv/l;)V
-
-    goto :goto_a
-
-    :cond_b
-    if-eqz v3, :cond_c
-
-    invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object v4
-
-    :goto_b
-    invoke-interface {v4}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v5
-
-    if-eqz v5, :cond_c
-
-    invoke-interface {v4}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v5
-
-    check-cast v5, LRg/F;
-
-    iget-object v5, v5, LRg/F;->b:Ljava/util/ArrayList;
-
-    sget-object v6, LRg/z;->a:LRg/z;
-
-    invoke-static {v5, v6}, Lrv/q;->A(Ljava/util/List;LFv/l;)V
-
-    goto :goto_b
-
-    :cond_c
-    if-eqz v3, :cond_d
-
-    invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object v4
-
-    :goto_c
-    invoke-interface {v4}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v5
-
-    if-eqz v5, :cond_d
-
-    invoke-interface {v4}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v5
-
-    check-cast v5, LRg/F;
-
-    iget-object v5, v5, LRg/F;->b:Ljava/util/ArrayList;
-
-    new-instance v6, LRg/x;
-
-    invoke-direct {v6, v1}, LRg/x;-><init>(LRg/s$a;)V
-
-    invoke-static {v5, v6}, Lrv/q;->A(Ljava/util/List;LFv/l;)V
-
-    goto :goto_c
-
-    :cond_d
-    if-eqz v3, :cond_e
-
-    invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object v4
-
-    :goto_d
-    invoke-interface {v4}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v5
-
-    if-eqz v5, :cond_e
-
-    invoke-interface {v4}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v5
-
-    check-cast v5, LRg/F;
-
-    iget-object v5, v5, LRg/F;->b:Ljava/util/ArrayList;
-
-    new-instance v6, LRg/y;
-
-    invoke-direct {v6, v1}, LRg/y;-><init>(LRg/s$a;)V
-
-    invoke-static {v5, v6}, Lrv/q;->A(Ljava/util/List;LFv/l;)V
-
-    goto :goto_d
-
-    :cond_e
-    if-eqz v1, :cond_f
-
-    iget-object p0, v1, LRg/s$a;->d:Ljava/lang/String;
-
-    :cond_f
-    if-nez p0, :cond_10
-
-    goto :goto_f
-
-    :cond_10
-    if-eqz v3, :cond_11
-
-    invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object p0
-
-    :goto_e
-    invoke-interface {p0}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v3
-
-    if-eqz v3, :cond_11
-
-    invoke-interface {p0}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v3
-
-    check-cast v3, LRg/F;
-
-    iget-object v3, v3, LRg/F;->b:Ljava/util/ArrayList;
-
-    new-instance v4, LRg/C;
-
-    invoke-direct {v4, v1}, LRg/C;-><init>(LRg/s$a;)V
-
-    invoke-static {v3, v4}, Lrv/q;->A(Ljava/util/List;LFv/l;)V
-
-    goto :goto_e
-
-    :cond_11
-    :goto_f
-    if-eqz p1, :cond_12
-
-    invoke-virtual {v0}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
-
-    move-result-object p0
-
-    :goto_10
-    invoke-interface {p0}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result p1
-
-    if-eqz p1, :cond_12
-
-    invoke-interface {p0}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object p1
-
-    check-cast p1, Lcom/xiaomi/cam/watermark/a;
-
-    iget-object v1, p1, Lcom/xiaomi/cam/watermark/a;->a:Ljava/nio/file/Path;
-
-    new-instance v3, Ljava/lang/StringBuilder;
-
-    const-string v4, "filterData: delete "
-
-    invoke-direct {v3, v4}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {v3, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v3}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v1
-
-    invoke-static {v2, v1}, LWr/e;->m(Ljava/lang/String;Ljava/lang/String;)V
-
-    iget-object p1, p1, Lcom/xiaomi/cam/watermark/a;->a:Ljava/nio/file/Path;
-
-    invoke-interface {p1}, Ljava/nio/file/Path;->toFile()Ljava/io/File;
-
-    move-result-object p1
-
-    const-string v1, "it.getFolderPath().toFile()"
-
-    invoke-static {p1, v1}, LGv/o;->g(Ljava/lang/Object;Ljava/lang/String;)V
-
-    invoke-static {p1}, LBv/l;->s(Ljava/io/File;)Z
-
-    goto :goto_10
-
-    :cond_12
-    invoke-virtual {v0}, Ljava/util/ArrayList;->clear()V
+    .locals 0
 
     return-void
 .end method
diff --git a/smali_classes6/com/xiaomi/camera/cloudfilter/entity/CloudFilterItem.smali b/smali_classes6/com/xiaomi/camera/cloudfilter/entity/CloudFilterItem.smali
index fd1d3626..eb8798c2 100644
--- a/smali_classes6/com/xiaomi/camera/cloudfilter/entity/CloudFilterItem.smali
+++ b/smali_classes6/com/xiaomi/camera/cloudfilter/entity/CloudFilterItem.smali
@@ -2139,64 +2139,11 @@
 .end method
 
 .method public final supported()Z
-    .locals 3
+    .locals 1
 
-    .line 1
-    sget-object v0, Lcom/xiaomi/camera/cloudfilter/constant/RenderType;->TYPE_LUT:Lcom/xiaomi/camera/cloudfilter/constant/RenderType;
-
-    invoke-virtual {p0}, Lcom/xiaomi/camera/cloudfilter/entity/CloudFilterItem;->getRenderType()Ljava/lang/String;
-
-    move-result-object v1
-
-    invoke-virtual {v0, v1}, Lcom/xiaomi/camera/cloudfilter/constant/RenderType;->isSupport(Ljava/lang/String;)Z
-
-    move-result v0
-
-    const/4 v1, 0x0
-
-    if-nez v0, :cond_0
-
-    sget-object v0, Lcom/xiaomi/camera/cloudfilter/constant/RenderType;->TYPE_ALGORITHM:Lcom/xiaomi/camera/cloudfilter/constant/RenderType;
-
-    .line 2
-    invoke-virtual {p0}, Lcom/xiaomi/camera/cloudfilter/entity/CloudFilterItem;->getRenderType()Ljava/lang/String;
-
-    move-result-object v2
-
-    .line 3
-    invoke-virtual {v0, v2}, Lcom/xiaomi/camera/cloudfilter/constant/RenderType;->isSupport(Ljava/lang/String;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_0
-
-    return v1
-
-    .line 4
-    :cond_0
-    sget-object v0, Landroid/os/Build;->DEVICE:Ljava/lang/String;
-
-    .line 5
-    invoke-static {v0}, LGv/o;->e(Ljava/lang/Object;)V
-
-    invoke-direct {p0, v0}, Lcom/xiaomi/camera/cloudfilter/entity/CloudFilterItem;->supported(Ljava/lang/String;)Z
-
-    move-result v2
-
-    if-eqz v2, :cond_1
-
-    invoke-direct {p0, v0}, Lcom/xiaomi/camera/cloudfilter/entity/CloudFilterItem;->unSupported(Ljava/lang/String;)Z
-
-    move-result p0
-
-    if-nez p0, :cond_1
-
-    const/4 p0, 0x1
-
-    return p0
+    const/4 v0, 0x1
 
-    :cond_1
-    return v1
+    return v0
 .end method
 
 .method public toString()Ljava/lang/String;
-- 
2.43.0

